File: mobile/ios/Sources/RoyalPMSMobile/Models/UserProfile.swift
import Foundation

struct UserProfile: Codable {
    let id: String
    let email: String
    let name: String
    let role: String
    let companyId: String?
    let phone: String?
    let photoUrl: String?

    enum CodingKeys: String, CodingKey {
        case id, email, name, role, phone
        case companyId = "company_id"
        case photoUrl  = "photo_url"
    }

    var canManageRooms: Bool {
        ["admin", "manager", "reception", "housekeeping", "maintenance"].contains(role)
    }

    var canManageTickets: Bool {
        ["admin", "manager", "maintenance", "reception"].contains(role)
    }
}

File: mobile/android/app/src/main/java/com/royalpms/mobile/data/models/UserProfile.kt
package com.royalpms.mobile.data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class UserProfile(
    val id: String,
    val email: String,
    val name: String,
    val role: String,
    @SerialName("company_id") val companyId: String? = null,
    val phone: String? = null,
    @SerialName("photo_url") val photoUrl: String? = null,
)

val UserProfile.canManageRooms get() =
    role in listOf("admin", "manager", "reception", "housekeeping", "maintenance")

val UserProfile.canManageTickets get() =
    role in listOf("admin", "manager", "maintenance", "reception")
